components/style/rule_tree/mod.rs

```diff
@@ -593,16 +597,29 @@ impl StrongRuleNode {
                     None => &self.get().first_child,
                 };
 
+                // We use `AqcRel` semantics to ensure the initializing writes
+                // in `node` are visible after the swap succeeds.
+                //
+                // TODO(emilio): Perhaps this could be `Relaxed`? When the CAS
+                // succeeds, the writes from the last thread that wrote into
+                // next_sibling_ptr are visible in this one. If we have the
+                // guarantee that _all_ memory written before that write is
+                // visible (which I think we do), that means that their
+                // initialization of the new node is visible into our thread
+                // too, so we shouldn't need this `AcqRel` semantics.
```

Comment:
If you do end up landing with this comment, please reference the discussion in this PR.